### ISSUE TITLE
Drop `kiam` from CAPA

### ISSUE THREAD

**Issue by alex-dabija** [giantswarm/roadmap#1781]
## Task

Drop `kiam` from CAPA.

## Background

Kiam is no longer needed because IAM Roles for Service Accounts (IRSA) is running on CAPA cluster as the official alternative.

## TODOs

- [x] Remove Kiam from the list of default apps on AWS;
- [x] Verify that all default applications are working correctly with IRSA;
- [x] Update the management clusters to drop Kiam;
    - [x] goat
    - [x] grizzly
    - [x]  golem
    - [x] thor
    - [x] thunder
- [x] Verify that Kiam is not running on the management clusters;

**Comment by primeroz** [comment]
When KIAM is removed i would like to test something , while investigating Workload Identity for CAPZ i found an issue where  , without any daemonset hijacking the metadata endpoint anymore, any pod can actually assume the identity of the node it is running on - see https://github.com/giantswarm/roadmap/issues/1728#issuecomment-1346913170

I checked on CAPA ( which i expected would have the same issue ) but at the moment this is not the case since KIAM will prevent that from happening. 

```
kubectl node-shell ip-10-0-102-120.eu-west-2.compute.internal

root@ip-10-0-102-120:/# aws sts get-caller-identity
{
    "UserId": "AROATQWT6AO7LXAWESMRZ:i-0cda1fa67c0cd18ee",
    "Account": "242036376510",
    "Arn": "arn:aws:sts::242036376510:assumed-role/nodes-machine-pool0-keep1/i-0cda1fa67c0cd18ee"
}

cat <<EOF | kubectl apply -f -
apiVersion: v1
kind: Pod
metadata:
  name: awscli
  namespace: default
spec:
  containers:
    - image: amazon/aws-cli:2.9.6
      name: cli
      command:
      - "sleep"
      - "3600"
  nodeSelector:
    kubernetes.io/os: linux
EOF

k exec -t -i awscli bash

 λ k exec -t -i awscli bash
kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl exec [POD] -- [COMMAND] instead.
bash-4.2# aws sts get-caller-identity

Unable to locate credentials. You can configure credentials by running "aws configure".
bash-4.2# 
```

blocked by kiam indeed

```
➜ k logs -n kube-system kiam-agent-2zmfj 
{"level":"info","msg":"configuring iptables","time":"2022-12-12T17:05:10Z"}
{"level":"info","msg":"started prometheus metric listener 0.0.0.0:9620","time":"2022-12-12T17:05:10Z"}
{"level":"info","msg":"listening :8181","time":"2022-12-12T17:05:10Z"}
{"addr":"100.64.3.45:54796","level":"error","method":"GET","msg":"error processing request: empty role","path":"/latest/meta-data/iam/security-credentials/","status":404,"time":"2022-12-12T17:40:20Z"}
```

So i wonder if , once KIAM is removed, the same problem will exist on CAPA

**Comment by alex-dabija** [comment]
Blocked until `external-dns` uses IRSA. See [issue](https://github.com/giantswarm/roadmap/issues/1640);

**Comment by primeroz** [comment]
The issue on AWS when KIAM is not running was confirmed by @whites11 https://gigantic.slack.com/archives/C02HLSDH3DZ/p1674550674581599

**Comment by alex-dabija** [comment]
It looks like `cert-manager` is not using IRSA: https://github.com/giantswarm/talanx/issues/95 .

**Comment by alex-dabija** [comment]
`cert-manager` is not using IRSA on a new cluster:
```
[container|default|alextest45-runtoolbox] /workdir | node:
 #: kubectl -n kube-system get serviceaccount cert-manager-controller -o yaml
apiVersion: v1
automountServiceAccountToken: true
kind: ServiceAccount
metadata:
  annotations:
    meta.helm.sh/release-name: cert-manager
    meta.helm.sh/release-namespace: kube-system
  creationTimestamp: "2023-01-30T12:16:00Z"
  labels:
    app: cert-manager
    app.kubernetes.io/component: controller
    app.kubernetes.io/instance: cert-manager
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: cert-manager
    application.giantswarm.io/team: team-cabbage
    giantswarm.io/service-type: managed
    helm.sh/chart: cert-manager-app-2.18.0
  name: cert-manager-controller
  namespace: kube-system
  resourceVersion: "3535"
  uid: 9516fdb6-bb29-4c13-b29e-5f6dfbd0c039
secrets:
- name: cert-manager-controller-token-9zd2x
```

`external-dns` is using IRSA on the same cluster:
```
[container|default|alextest45-runtoolbox] /workdir | node:
 #: kubectl -n kube-system get serviceaccount external-dns -o yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    eks.amazonaws.com/role-arn: alextest45-Route53Manager-Role
    meta.helm.sh/release-name: external-dns
    meta.helm.sh/release-namespace: kube-system
  creationTimestamp: "2023-01-30T12:15:50Z"
  labels:
    app: external-dns
    app.kubernetes.io/instance: external-dns
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: external-dns
    app.kubernetes.io/version: v0.11.0
    application.giantswarm.io/team: team-cabbage
    giantswarm.io/service-type: managed
    helm.sh/chart: external-dns-app-2.23.2
  name: external-dns
  namespace: kube-system
  resourceVersion: "3388"
  uid: 3fdc400f-0f60-44e1-a8c9-54d6822a73e8
secrets:
- name: external-dns-token-9gfwx
```

**Comment by tuladhar** [comment]
Cert manager seems to already support IRSA:
- https://cert-manager.io/docs/configuration/acme/dns01/route53/#eks-iam-role-for-service-accounts-irsa

**Comment by tuladhar** [comment]
On hold, until cert-manager-app supports IRSA.

**Comment by alex-dabija** [comment]
`cert-manager` has support for IRSA in the latest version (see [comment](https://github.com/giantswarm/talanx/issues/95#issuecomment-1411935218)).

**Comment by alex-dabija** [comment]
Blocked until IRSA is enabled for `cert-manager`. See [issue](https://github.com/giantswarm/roadmap/issues/1981).

**Comment by AndiDog** [comment]
It seems that CAPA [uses](https://github.com/giantswarm/cluster-api-provider-aws-app/blob/master/helm/cluster-api-provider-aws/templates/apps_v1_deployment_capa-controller-manager.yaml) the kiam annotation (`iam.amazonaws.com/role: defaultARN:=""`) and we have to migrate it as well. That would be a blocker.

**Comment by calvix** [comment]
> It seems that CAPA [uses](https://github.com/giantswarm/cluster-api-provider-aws-app/blob/master/helm/cluster-api-provider-aws/templates/apps_v1_deployment_capa-controller-manager.yaml?rgh-link-date=2023-03-20T09%3A41%3A14Z) the kiam annotation (`iam.amazonaws.com/role: defaultARN:=""`) and we have to migrate it as well. That would be a blocker.

CAPA doesnt use kiam

**Comment by AndiDog** [comment]
Seems a leftover. CAPA uses the secret `capa-manager-bootstrap-credentials` with static AWS credentials.

**Comment by calvix** [comment]
golem grizzly and goat got updated to version without kiam, will leave until next week and update the rest of our MCs

**Comment by calvix** [comment]
done

**Comment by LolloneS** [comment]
Would it be possible to remove `kiam` from `anzu` as well? Thanks.

**Comment by fiunchinho** [comment]
@LolloneS kiam has been removed from `anzu`.
